### ISSUE TITLE
Drawn mask opacity is kind of reversed.

### ISSUE THREAD

**Issue by infjb** [darktable-org/darktable#3772]
The precompiled 3.0.0rc2 works like expected. 

I compiled my own version to test the opencl fix, which works great. I compiled darktable on Windows 10 using mingw64. The version string says:

`3.0.0rc2+101~g12a3e26fc-dirty`

So I don't know if this is a problem of my self compiled binary or if a bug got introduced over the last days.

When using drawn masks, the opacity works kind of backwards. When using the scroll wheel while holding the `CRTL` key, the opacity of the mask changes. The opacity is chown above the picture in percent, the overlay of the  drawn area on the screen changes between a dark and a light grey.

100% opacity should result in the maximum effect of the module and the darkest overlay, 0% opacity should result in the minimum effect an the lightest overlay. 

With my self compiles binary it is the other way round. 100% opacity are resulting in the lightest overlay and the minimum effect. 0% opacity are giving a medium gray overlay an a medium effect. The full effect is unreachable with a drawn mask.

A mask drawn with version 3.0.0rc2 or older still results in the maximum effect. Only newly drawn masks are effected.

You can easily test it with the contrast, brightness and saturation module or when using the blur tool of the retouch module.

It might be an artifact of my mingw64 setup.

**Comment by arigit** [comment]
@TurboGit there seems to be indeed a problem with mask behavior in DT 3.0.x (current master head) that make use un-intuitive (looks like a bug). 
Tested on Ubuntu/linux but based on the comment above this applies to windows as well.

The problems seem to be in new Shape creation. It appears there are two different problems:

1) when adding a new shape via any module the Shape Opacity is set to zero percent which is very confusing (this used to be 100% before), also when drawing the shape, the drawing (e.g brushing) confusingly behaves as if the opacity were 100% as in previous versions, this is most notable with the Brush, in the past, when "brushing" with the Mask Display turned on, one would see the dark "trail" on the image left by the brush, the higher the opacity the darker the trail, and this seems to be reverted now: the new '0% default opacity' leaves the darker trail. Clearly this is a bug

2) when inserting a new shape, after the creation is done, the shape insertion 'mini toolbar' inside the module continues to show "shape insertion state" as if one wanted to endlessly keep adding shapes, instead of the previous behavior where one would see the newly drawn shape outline. In the past, I believe the retouch tool used to offer and "endless" add-shape mode that was useful when one was retouching multiple spots, but in general the "endless add-shape mode" is confusing when applied to other modules and it seems to be applied by default now for all modules. 

With both issues happening adding Drawn Masks can get really confusing in DT

**Comment by arigit** [comment]
RE #1 above, 0% opacity used as default for new shapes, I could reproduce this with with DT 3.0 final, however somehow while testing different combinations (changing opacity before and after completing shape insertion, etc) somehow this changed back to normal: now new shapes get inserted with 100% opacity be default which makes more sense. I tried to find what was the combination that triggered DT to 'remember' the new default, there must be one, but I didn't succeed. At least now pleople can see that this issue can be reverted :)

RE #2, still present in DT3.0 final - it's most visible in the retouch module. I compared with DT2.6, and there in order to trigger the mode of "continuous shape additions" one hand to Ctrl-click the shape that was wanted. After completing all additions, one final click in the shape (mini-toolbar in the module) and the module goes back to normal. Looks like DT3 changed the default, I guess by mistake is it is confusing even for long time users (it was reported a few times already)
